src/Sarif.Multitool/Rules/RuleId.cs

```diff
@@ -16,5 +16,6 @@ public static class RuleId
         public const string EndColumnMustNotBeLessThanStartColumn = "SARIF1013";
         public const string UriBaseIdRequiresRelativeUri = "SARIF1014";
         public const string UriMustBeAbsolute = "SARIF1015";
+        public const string ContextRegionRequiresRegion = "SARIF1016";
```

Comment:
>SARIF1016 [](start = 59, length = 9)

I have a funky idea for you which is to at least consider the possibility that the relevant SARIF spec section comprise the rule id. So SARIF.3.24.5 for this case (the analysis that addresses subtleties in the physicalLocation.contextRegion object. NOW: of course, the spec is malleable at this point in time. so this may simply be a lousy idea. Or, we might proceed with writing validators until the spec is finalized, then do a scrub to convert to final section numbers. Don't forget, we have the hierarchical string to allow some flexibility as well in creating sub-checks that roll up into a master rule id. SARIF.3.24/5, for example, would point to a master rule, dedicated to validating the physicalLocation object, the sub-rule of which validates the plc.contextRegion is handled properly. Something to ponder.

Reply:
I worry about those ids breaking in the next version of the spec. Even an editorial section reordering would break them, let alone new content ("3.25 metrics object").

---
In reply to: [238501875](https://github.com/Microsoft/sarif-sdk/pull/1142#discussion_r238501875) [](ancestors = 238501875)